slither/tools/read_storage/read_storage.py

```diff
@@ -111,9 +129,46 @@ def get_storage_layout(self) -> None:
                 elif isinstance(type_, ArrayType):
                     elems = self._all_array_slots(var, contract, type_, info.slot)
                     tmp[var.name].elems = elems
-
+        tmp = self.get_unstructured_layout(tmp)
         self._slot_info = tmp
 
+    def get_unstructured_layout(self, tmp: dict) -> dict:
+        for _, var in self.constant_slots:
+            var_name = var.name
+            try:
+                exp = var.expression
+                if isinstance(exp, CallExpression) and str(exp.called) == "keccak256(bytes)":
+                    exp = exp.arguments[0]
+                if isinstance(
+                    exp,
+                    (BinaryOperation, UnaryOperation, Identifier, TupleExpression, TypeConversion),
+                ):
+                    exp = ConstantFolding(exp, "bytes32").result()
```

Comment:
The `NotConstant` error is not being handling here

Reply:
If the expression is not constant then I suppose the variable in question can't be a constant storage slot, so we can skip it by adding `NotConstant` to the except clause on line 171:
```python
    def get_unstructured_layout(self) -> Dict[str, SlotInfo]:
        tmp: Dict[str, SlotInfo] = {}
        for _, var in self.constant_slots:
            var_name = var.name
            try:
                exp = var.expression
                if isinstance(exp, CallExpression) and str(exp.called) == "keccak256(bytes)":
                    exp = exp.arguments[0]
                if isinstance(
                    exp,
                    (BinaryOperation, UnaryOperation, Identifier, TupleExpression, TypeConversion),
                ):
                    exp = ConstantFolding(exp, "bytes32").result()
                ...
            except (TypeError, NotConstant):
                continue
```